tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml

```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
+  #Checking first Date
+    - Type: get_attribute
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.first_day$
+      Greps:
+        - var: FIRST_SCHEDULED_DAY
+          attr: datetime
+          condition: nempty
+          match: "(.*)"
+    - Type: get_day
+      Timestamp: $AND_CLI_FIRST_SCHEDULED_DAY$
+      ResultVar: DAY
+    # choosing Date
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.select_location_drop_down$
+      Id: $PAGE.employee_health_home_page.calendar_right_button$
+    - Type: wait_for
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.calendar_month_view$
+    # checking first date introduced
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.select_location_option$
-    - Type: generate_unique_name
-      Name: "I have something to say about my health."
-      ResultVar: PAIN_DESCRIPTION
-    - Type: send_keys
-      Strategy: id
-      Id: $PAGE.employee_health_telehealth_request_form.comments$
-      Value: $AND_CLI_PAIN_DESCRIPTION$
-    # choosing Date and time
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_right_button$
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1 
+    # Click on the View upcoming appointments
     - Type: wait_for
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_month_view$
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_day_button$
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
+    # getting second day
+    - Type: get_attribute
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.second_day$
+      Greps:
+        - var: SECOND_SCHEDULED_DAY
+          attr: datetime
+          condition: nempty
+          match: "(.*)"
+    - Type: get_day
+      Timestamp: $AND_CLI_SECOND_SCHEDULED_DAY$
+      ResultVar: DAY
+    # checking second day
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_hour_button$
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.create_event_button$
-  # Verify if the appointment is showing on the list
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
```

Comment:
is this sleep needed?

Reply:
yes, when I use wait_for  the test failure trying to click on View upcoming appointment button.